### PR TITLE
distsql/select_result_test: migrate test-infra to testify

### LINKED ISSUE #26156: migrate test-infra to testify for `distsql` pkg

### PR DESCRIPTION
**Pull request by tomatopunk**
### What problem does this PR solve?

Issue Number: ref #26156 

Split from folder distsql test

- [x] distsql/select_result_test
- [ ] distsql/distsql_test
- [ ] distsql/request_build_test

```release-note
None
```

### CODE REVIEW

File: distsql/main_test.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package distsql
+
+import (
+	"github.com/pingcap/tidb/util/testbridge"
+	"go.uber.org/goleak"
+	"testing"
```
Comment (tisonkun) [inline distsql/main_test.go:6]: goimports failed. see also https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/17617/pipeline

File: distsql/select_result_test.go [hunk 2]
```diff
@@ -15,8 +15,9 @@ package distsql
 
 import (
 	"context"
+	"github.com/stretchr/testify/require"
+	"testing"
```
Comment (tisonkun) [inline distsql/select_result_test.go:19]: ditto

File: distsql/main_test.go (hunk 1 above)
Comment (tomatopunk) [reply distsql/main_test.go]: thks. already resolve!

File: distsql/select_result_test.go [hunk 3]
```diff
@@ -15,38 +15,69 @@ package distsql
...
 	"github.com/pingcap/tidb/store/copr"
 	"github.com/pingcap/tidb/util/execdetails"
 	"github.com/pingcap/tidb/util/mock"
 	"github.com/pingcap/tipb/go-tipb"
+
+	"github.com/stretchr/testify/require"
 )
 
-func (s *testSuite) TestUpdateCopRuntimeStats(c *C) {
+func TestUpdateCopRuntimeStats(t *testing.T) {
```
Comment (tisonkun) [inline distsql/select_result_test.go:30]: As they share the same `sr` variable, I don't think you should run them in sub test but directly run in `TestUpdateCopRuntimeStats`.

Comment (tomatopunk) [reply distsql/select_result_test.go:30]: I feel more like three steps in one somethings,so i put sr outside,which seems to destroy the `independence of the test`. let me fix it

> As they share the same `sr` variable, I don't think you should run them in sub test but directly run in `TestUpdateCopRuntimeStats`.

I feel more like three steps in one somethings,so i put sr outside,which seems to destroy the `independence of the test`. let me fix it

Comment (tisonkun) [reply distsql/select_result_test.go:30]: ... also pay attention on parallel. `t.Parallel` is a stmt that has its own effect, not a magic that you may or may not place it somewhere. For example, it is confused you call `t.Parallel` in both parent test and its child.

For common setup logic, see also 

https://github.com/pingcap/tidb/blob/dd5546dfd0ad8376b3c7188da7c19d8c3b1818ab/util/admin/admin_test.go#L83-L84

Comment (tomatopunk) [reply distsql/select_result_test.go:30]: thks, has already been updated

File: distsql/main_test.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package distsql
```
Comment (tisonkun) [inline distsql/main_test.go:1]: We need a license header here, see also other files.

File: distsql/select_result_test.go [hunk 5]
```diff
@@ -15,38 +15,46 @@ package distsql
...
 
 	sr.copPlanIDs = []int{sr.rootPlanID}
-	c.Assert(ctx.GetSessionVars().StmtCtx.RuntimeStatsColl, NotNil)
-	c.Assert(len(sr.selectResp.GetExecutionSummaries()), Equals, len(sr.copPlanIDs))
+	require.NotNil(t, ctx.GetSessionVars().StmtCtx.RuntimeStatsColl)
+	require.Equal(t, len(sr.selectResp.GetExecutionSummaries()), len(sr.copPlanIDs))
+
 	sr.updateCopRuntimeStats(context.Background(), &copr.CopRuntimeStats{ExecDetails: execdetails.ExecDetails{CalleeAddress: "callee"}}, 0)
-	c.Assert(ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetOrCreateCopStats(1234, "tikv").String(), Equals, "tikv_task:{time:1ns, loops:1}")
+	require.Equal(t, ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetOrCreateCopStats(1234, "tikv").String(), "tikv_task:{time:1ns, loops:1}")
```
Comment (tisonkun) [inline distsql/select_result_test.go:59]:
```suggestion
	require.Equal(t, "tikv_task:{time:1ns, loops:1}", ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.GetOrCreateCopStats(1234, "tikv").String())
```

be aware of the position of `expected` and `actual`

File: distsql/main_test.go (hunk 4 above)
Comment (tomatopunk) [reply distsql/main_test.go:14]: > We need a license header here, see also other files.

thanks,fixed and update

File: distsql/select_result_test.go (hunk 5 above)
Comment (tomatopunk) [reply distsql/select_result_test.go]: thanks,fixed and update

File: distsql/main_test.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 PingCAP, Inc.
```
Comment (tisonkun) [inline distsql/main_test.go:1]:
```suggestion
// Copyright 2021 PingCAP, Inc.
```

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
/cc @tiancaiamao @xhebox 

PTAL

**Review by xhebox — APPROVED** [review]

**Review by mmyj — APPROVED** [review]
lgtm

**Pull request merged**